pkg/cmd/run/list/list.go

```diff
@@ -137,7 +137,17 @@ func listRun(opts *ListOptions) error {
 		tp.EndRow()
 	}
 
+	IDToName := make(map[int64]string)
+
 	for _, run := range runs {
+		if _, ok := IDToName[run.WorkflowID]; !ok {
+			name, err := run.GetWorkflowName(client, baseRepo, run.WorkflowID)
```

Comment:
1. Could we push this into `shared.GetRuns` so that it's also available to JSON export and that the caller doesn't have to be concerned with performing API lookups?
2. I'm concerned about the number of API lookups here. This additional lookup causes N additional API requests on each `gh run list`, where N is the number of distinct workflows. Is there a way we could consolidate this, for example by doing 1 API request to fetch the runs and 1 API request to fetch all the workflows?

Reply:
1. Yes, this should be available for all methods getting workflow runs
2. Yes, we're now getting all the workflow sin 1 run,